client/packages/invoices/src/Returns/modals/InboundReturn/ReturnQuantitiesTable.tsx

```diff
@@ -15,35 +21,76 @@ export const QuantityReturnedTableComponent = ({
     line: Partial<GeneratedInboundReturnLineNode> & { id: string }
   ) => void;
 }) => {
+  // what to do here if there are multiple items? Some might have PV and others not...
+  // const { packVariantExists } = usePackVariant(item?.id || '', null);
+
   const columns = useColumns<GeneratedInboundReturnLineNode>(
     [
       'itemCode',
       'itemName',
       // 'itemUnit', // not implemented for now
       // 'location',
-      'batch',
-      'expiryDate',
       [
-        'numberOfPacks',
+        'batch',
         {
-          label: 'label.pack-quantity-issued',
           width: 125,
-          accessor: ({ rowData }) => rowData.numberOfPacksIssued,
+          accessor: ({ rowData }) => rowData.batch ?? '',
+          setter: updateLine,
+          Cell: TextInputCell,
+        },
+      ],
+      [
+        expiryInputColumn,
+        {
+          width: 150,
+          setter: l =>
+            updateLine({
+              ...l,
+              expiryDate: l.expiryDate
+                ? Formatter.naiveDate(new Date(l.expiryDate))
+                : null,
+            }),
+        },
+      ],
+      [
+        'packSize',
+        {
+          width: 100,
+          setter: updateLine,
+          Cell: PackSizeInputCell,
         },
       ],
+      // TODO: implement pack variant
+      // getColumnLookupWithOverrides('packSize', {
+      //   Cell: PackUnitEntryCell,
+      //   setter: updateLine,
+      //   ...(packVariantExists
+      //     ? {
+      //         label: 'label.unit-variant-and-pack-size',
+      //         minWidth: PACK_VARIANT_ENTRY_CELL_MIN_WIDTH,
+      //       }
+      //     : { label: 'label.pack-size' }),
+      // }),
+      ...(lines.some(l => l.numberOfPacksIssued !== null) // if any line has a value, show the column
+        ? ([
+            [
+              'numberOfPacks',
+              {
+                label: 'label.pack-quantity-issued',
+                width: 110,
+                accessor: ({ rowData }) => rowData.numberOfPacksIssued ?? '--',
+                Cell: BasicCell,
+              },
+            ],
+          ] as ColumnDescription<GeneratedInboundReturnLineNode>[])
+        : []),
```

Comment:
Is there a better way to do this piece, only showing the Qty issued column if the data is present?

Reply:
Can't think of a better way 🤷‍♂️

Reply:
I played around with this syntax a bit, I've never been happy with how it looks. Ideally useColumns should just skip a null. Would width = 0 work (or is that hacky). Anyways ok with what you have here for now

Reply:
Haha I mean this way feels a bit hacky to me too - you're width 0 idea feels kind of cleaner to me, but it doesn't hide the column heading (it just overlaps the next column 😅)